Swift/Adhan/Adhan.swift

```diff
@@ -403,6 +408,35 @@ public struct Qibla {
     }
 }
 
+/* Sunnah times for a location and date using the given prayer times.
+All prayer times are in UTC and should be display using a DateFormatter that
+has the correct timezone set. */
+public struct SunnahTimes {
+    public let middleOfTheNight: Date
+    public let lastThirdOfTheNight: Date
+
+    public init?(from prayerTimes: PrayerTimes) {
+        let cal: Calendar = .gregorianUTC
+        let tomorrow = cal.date(byAdding: .day, value: 1, to: prayerTimes.fajr)!
+        
+        guard let nextDayPrayers = PrayerTimes(
+            coordinates: prayerTimes.coordinates,
+            date: cal.dateComponents([.year, .month, .day], from: tomorrow),
+            calculationParameters: prayerTimes.calculationParameters) else {
+                // unable to determine tomorrow prayer times
+                return nil
+        }
+        
+        let seconds = cal.dateComponents([.second], from: prayerTimes.maghrib, to: nextDayPrayers.fajr).second!
```

Comment:
I actually meant using Date's timeIntervalSinceDate function which gives you an NSTimeInterval value of the difference in seconds. By using date components it has to do expensive calendrical calculations, when using timeIntervalSinceDate it merely subtracts two doubles which is much more efficient.